aws/csm/metricChan_test.go

```diff
@@ -0,0 +1,72 @@
+package csm
+
+import (
+	"testing"
+)
+
+func TestMetricChanPush(t *testing.T) {
+	ch := newMetricChan(5)
+	defer close(ch.ch)
+
+	pushed := ch.Push(metric{})
+	if !pushed {
+		t.Errorf("expected metrics to be pushed")
+	}
+
+	if e, a := 1, len(ch.ch); e != a {
+		t.Errorf("expected %d, but received %d", e, a)
+	}
+}
+
+func TestMetricChanPauseContinue(t *testing.T) {
```

Comment:
I think it will be helpful to have a race condition test with multiple writers to the metric chan in parallel, pausing the chan, and unpausing it.  Would also be really helpful to make sure that there are no blocked writers to the channel if Pause is called.